test/cri-containerd/test-images/unordered_tar/main.go

```diff
@@ -0,0 +1,229 @@
+	// converted into the image tar at the end.
+	tempDirPath, err := ioutil.TempDir("", "imagecreator-imagedir")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temporary directory")
+	}
+	defer os.RemoveAll(tempDirPath)
+
+	sha256dirPath := filepath.Join(tempDirPath, "blobs", "sha256")
+	err = os.MkdirAll(sha256dirPath, 0777)
+	if err != nil {
+		return errors.Wrap(err, "failed to create blobs dir")
+	}
+
+	// copy all layer tar as blobs
+	layerBlobs := []ocispec.Descriptor{}
+	for _, layerTar := range layerTars {
+		llen, dgst, err := createBlobFromTar(layerTar, sha256dirPath)
+		if err != nil {
+			return errors.Wrap(err, "layer content write failed")
+		}
+		layerBlobs = append(layerBlobs, ocispec.Descriptor{
+			MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip",
+			Digest:    dgst,
+			Size:      llen,
+		})
+	}
+
+	// create image config blob
+	clen, cdgst, err := createBlobFromStruct(sha256dirPath, createMinimalConfig(layerBlobs))
+	if err != nil {
+		return errors.Wrap(err, "failed to create config blob")
+	}
+
+	// create manifest blob
+	var manifest ocispec.Manifest
+	manifest.SchemaVersion = 2
+	manifest.Config.Size = clen
+	manifest.Config.Digest = cdgst
+	manifest.Config.MediaType = "application/vnd.docker.container.image.v1+json"
+	manifest.Layers = layerBlobs
+	mlen, mdgst, err := createBlobFromStruct(sha256dirPath, manifest)
+	if err != nil {
+		return errors.Wrap(err, "failed to crate blob for manifest")
+	}
+
+	// create index file
+	var index ocispec.Index
+	index.SchemaVersion = 2
+	index.Manifests = append(index.Manifests, ocispec.Descriptor{
+		MediaType: "application/vnd.docker.distribution.manifest.v2+json",
+		Digest:    mdgst,
+		Size:      mlen,
+	})
+
+	indexJson, err := json.Marshal(index)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal index json")
+	}
+
+	findex, err := os.Create(filepath.Join(tempDirPath, "index.json"))
+	if err != nil {
+		return errors.Wrap(err, "failed to create index file")
+	}
+	_, err = findex.Write(indexJson)
+	if err != nil {
+		return errors.Wrap(err, "failed to write index.json")
+	}
+
+	// create oci layout file
+	if err = createOciLayoutFile(tempDirPath); err != nil {
+		return err
+	}
+
+	// create tar from the image
+	tarCmd := exec.Command("tar", "-C", tempDirPath, "-cf", "testimage.tar", ".")
+	if err = tarCmd.Run(); err != nil {
+		return errors.Wrap(err, "image tar creation failed")
+	}
+	return nil
+}
+
+func main() {
+	tempDir, err := ioutil.TempDir("", "imagecreator-layerdir")
+	if err != nil {
+		fmt.Printf("failed to create temp dir: %s\n", err)
+		return
+	}
+	defer os.RemoveAll(tempDir)
+
+	layerTars, err := createUnorderedTars(tempDir)
+	if err != nil {
+		fmt.Printf("failed to create unordered tars: %s\n", err)
+		return
+	}
+
+	if err = createImageFromLayerTars(layerTars); err != nil {
+		fmt.Printf("failed to create image: %s\n", err)
+		return
```

Comment:
I'd recommend to use `log.Fatalf("something exploded: %s", err)` on the error cases so at least running this will return exit code 1 if anything fails (and you save the 'print to stdout' -> return flow three places).

Reply:
Fixed.